jib-core/src/main/java/com/google/cloud/tools/jib/api/JibContainerBuilder.java

```diff
@@ -167,36 +170,82 @@ public JibContainerBuilder addLayer(List<Path> files, String pathInContainer) th
   /**
    * Adds a layer (defined by a {@link LayerConfiguration}).
    *
+   * <p>Deprecated. Use {@link #addFileEntriesLayer(FileEntriesLayer)}.
+   *
    * @param layerConfiguration the {@link LayerConfiguration}
    * @return this
    */
+  @Deprecated
   public JibContainerBuilder addLayer(LayerConfiguration layerConfiguration) {
-    layerConfigurations.add(layerConfiguration);
+    layerConfigurations.add(layerConfiguration.toFileEntriesLayer());
+    return this;
+  }
+
+  /**
+   * Adds a layer (defined by a {@link FileEntriesLayer}).
+   *
+   * @param layer the {@link FileEntriesLayer}
+   * @return this
+   */
+  public JibContainerBuilder addFileEntriesLayer(FileEntriesLayer layer) {
+    layerConfigurations.add(layer);
     return this;
   }
 
   /**
    * Sets the layers (defined by a list of {@link LayerConfiguration}s). This replaces any
    * previously-added layers.
    *
+   * <p>Deprecated. Use {@link #setFileEntriesLayers(List)}.
+   *
    * @param layerConfigurations the list of {@link LayerConfiguration}s
    * @return this
    */
+  @Deprecated
   public JibContainerBuilder setLayers(List<LayerConfiguration> layerConfigurations) {
-    this.layerConfigurations = new ArrayList<>(layerConfigurations);
+    this.layerConfigurations =
+        layerConfigurations
+            .stream()
+            .map(LayerConfiguration::toFileEntriesLayer)
+            .collect(Collectors.toList());
```

Comment:
should this call `setFileEntriesLayers` with the final result of the `stream.collect`?

Reply:
Makes sense. I just wanted to avoid creating an additional `ArrayList` in `setFileEntriesLayers`, but that's really negligible.

Also updated `addLayer()` to call `addFileEntriesLayer()` similarly.